### PR TITLE
feat: add support for local storage filters in use url filters

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
add support for local storage filters in use url filters

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Common/Hooks/useUrlFilters/useUrlFilters.ts [hunk 1]
```diff
@@ -126,6 +127,7 @@ const useUrlFilters = <T = string, K = unknown>({
 
     const clearFilters = () => {
         history.replace({ search: '' })
+        localStorage.setItem(localStorageKey, '')
```
Comment (eshankvaish) [inline src/Common/Hooks/useUrlFilters/useUrlFilters.ts:130]: shouldn't we check if the localStorageKey exists? Better would be create a internal function for setting/getting/removing the item

File: src/Common/Hooks/useUrlFilters/useUrlFilters.ts [hunk 2]
```diff
@@ -143,10 +145,31 @@ const useUrlFilters = <T = string, K = unknown>({
                 searchParams.delete(key)
             }
         })
+        localStorage.setItem(localStorageKey, JSON.stringify(parseSearchParams(searchParams)))
```
Comment (eshankvaish) [inline src/Common/Hooks/useUrlFilters/useUrlFilters.ts:148]: leave a comment mentioning that we are skipping the primary filters like page number, sorting etc.

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Common/Hooks/useUrlFilters/useUrlFilters.ts [hunk 3]
```diff
@@ -143,10 +145,31 @@ const useUrlFilters = <T = string, K = unknown>({
...
+        if (!localStorageKey) {
+            return
+        }
+        if (
+            Object.keys(parsedParams).some(
+                (key) =>
+                    (Array.isArray(parsedParams[key]) && parsedParams[key].length) ||
+                    (typeof parsedParams[key] === 'string' && !!parsedParams[key]),
+            )
+        ) {
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/Common/Hooks/useUrlFilters/useUrlFilters.ts:163]: Not clearing local storage so that we can store the filters and use for next visit.

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — DISMISSED** [review]

File: src/Common/Hooks/useUrlFilters/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const setItemInLocalStorageIfKeyExists = (localStorageKey: string, value: string) => {
```
Comment (RohitRaj011) [inline src/Common/Hooks/useUrlFilters/utils.tsx:1]: future scope can be making a LocalStorage Util class

**Review by RohitRaj011 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**